core/v2.go

```diff
@@ -2586,6 +2588,57 @@ func (core *Core) v2API() *route.Router {
 	})
 	// }}}
 
+	/* This following route is an out-of-band task that self-restores SHIELD if chosen
+	to during the initialization of SHIELD */
+	r.Dispatch("POST /v2/bootstrap/restore", func(r *route.Request) { // {{{
+		if core.IsNotSystemAdmin(r) {
+			return
+		}
+		file, header, err := r.Req.FormFile("archive")
+
+		fmt.Printf(header.Filename)
+
+		f, err := os.Create("/tmp/bootstrap.restore")
```

Comment:
See if we can use a tempfile library like https://golang.org/pkg/io/ioutil/#TempFile